### PR TITLE
Add a promotion page for Premium

### PR DESCRIPTION
**Pull request by Vinnl**
# New feature description

This adds a new interstitial at `/premium` that existing users can be directed to to outline the advantages of Premium. It largely copy-pastes the current landing page, since the designs are so similar.

# Screenshot (if applicable)

Mobile:

image

Desktop:

image

# How to test

Visit `/premium`.

# Checklist

- [x] l10n dependencies have been merged, if any. https://github.com/mozilla-l10n/fx-private-relay-l10n/pull/47
- [ ] All acceptance criteria are met. - Almost, except that there should also be an upgrade button in the header, which is a pain. I've asked about that and a couple of other implementation details in the ticket.
- [ ] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/static/scss/libs/protocol/css/includes/tokens/dist/index.scss`). - Not quite, but I tried to stay consistent with the class names already used on the landing page, to avoid `c-`s and `c-`-less classes interspersed.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by Vinnl** [comment]
Added a small change that, instead of redirecting people to the homepage if Premium is not available in their country, it adds a small message and disables the CTAs.

File: privaterelay/templates/premium-promo.html [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                                    <li>{% ftlmsg 'landing-pricing-premium-feature-4' %}</li>
+                                </ul>
+                            </div>
+                            <a
+                                class="mzp-c-button mzp-t-product mzp-t-secondary mzp-t-lg js-purchase-premium"
+                                href="{% premium_subscribe_url accept_language country_code %}"
+                                target="_blank" rel="noopener noreferrer"
+                                data-ga="send-ga-pings"
+                                data-event-category="Purchase Button"
+                                data-event-label="newlanding-plans-button"
```
Comment (maxxcrawford) [inline privaterelay/templates/premium-promo.html:234]: Should this be more page specific? Is it okay that it shares the same tags with the homepage?

File: static/scss/partials/premium-promo.scss [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+.perks-lead {
+    @include text-body-md;
+    display: block;
+
+    @media screen and #{$mq-md} {
+        margin: 0 auto;
+    }
+    width: 75%;
```
Comment (maxxcrawford) [inline static/scss/partials/premium-promo.scss:32]: nit: move `width` above the media query for easier reading

File: static/scss/partials/premium-promo.scss [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    flex-direction: column;
+    gap: $spacing-2xl;
+    align-items: center;
+    padding: $spacing-2xl 0;
+
+    @media screen and #{$mq-md} {
+        flex-direction: row;
+        padding: $spacing-2xl;
+
+        &:nth-child(odd) {
```
Comment (maxxcrawford) [inline static/scss/partials/premium-promo.scss:46]: Looking at the design, this should actually be `even`. The Figma file shows the first content section on desktop has the image on the left and the copy on the right, while the demo site has it flipped.

File: privaterelay/templates/premium-promo.html [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+            <p class="perk-body">
+                {% ftlmsg 'premium-promo-perks-perk-unlimited-body' %}
+            </p>
+            {% if premium_available_in_country %}
+                <a
+                    class="mzp-c-button mzp-t-product "
+                    href="{% premium_subscribe_url accept_language country_code %}"
+                    data-ga="send-ga-pings"
+                    data-event-category="Purchase Button"
+                    data-event-label="premium-promo-perk-cta"
```
Comment (maxxcrawford) [inline privaterelay/templates/premium-promo.html:69]: Should we differentiate between each perk section?

**Review by maxxcrawford — APPROVED** [review]
See notes below. I will leave this for you to merge in! Great work! 

I'm also including a seperate PR to yours that you can incorporate. It addresses these changes along with some nit fixes on the use-case selector.

File: static/scss/partials/premium-promo.scss (hunk 3 above)
Comment (Vinnl) [reply static/scss/partials/premium-promo.scss]: Good eye, thanks for the fix!

File: privaterelay/templates/premium-promo.html (hunk 1 above)
Comment (Vinnl) [reply privaterelay/templates/premium-promo.html]: Well, I just happened to learn how we do analytics (i.e. how the `data-ga` attributes work) yesterday, but _after_ I submitted this PR. :) Fixed in 69c24ad7497f6795b2440a337f1dff2f25f0f7f8.

File: privaterelay/templates/premium-promo.html (hunk 4 above)
Comment (Vinnl) [reply privaterelay/templates/premium-promo.html]: Likewise: e769db18

File: static/scss/partials/premium-promo.scss (hunk 2 above)
Comment (Vinnl) [reply static/scss/partials/premium-promo.scss]: Heh, how did that get there :sweat_smile: 4c63f4ba

**Pull request merged**